### ISSUE TITLE
Incorrect arrays validations in Dry::Struct attributes when used in schemas

### ISSUE THREAD

**Issue by semenovDL** [dry-rb/dry-validation#395]
Arrays inside Dry::Struct attributes works incorrectly when used in validation schemas.

```ruby
require 'dry-types'
require 'dry-struct'
require 'dry-validation'

Dry::Validation.load_extensions(:struct)

class S < Dry::Struct
  attribute :amounts, Dry::Types['strict.array'].of(Dry::Types['strict.int'])
end

schema = Dry::Validation.Schema do
  required(:plain).schema do
    required(:amounts).each(Dry::Types['strict.int'])
  end
  required(:struct).schema(S)
end

puts schema.call(
  plain: { amounts: [] },
  struct: { amounts: [] }
).errors

# expected: 
# {}
# actual:
# {:struct=>{:amounts=>["must be filled"]}}

puts schema.call(
  plain: { amounts: ['wrong', 123] },
  struct: { amounts: ['wrong', 123] }
).errors

# expected:
# {:plain=>{:amounts=>{0=>["must be Integer"]}}, :struct=>{:amounts=>{0=>["must be Integer"]}}}

# actual:
# {:plain=>{:amounts=>{0=>["must be Integer"]}}}
```

**Comment by semenovDL** [comment]
Fix it in [#396](https://github.com/dry-rb/dry-validation/pull/396)

**Comment by solnic** [comment]
Support for dry-struct (at least in this form) will be removed in 1.0.0 so this won’t be addressed.
